src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Implementation/Deployments/RemoveAzureDeploymentCmdlet.cs

```diff
@@ -0,0 +1,76 @@
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.Components;
+using Microsoft.Azure.Commands.ResourceManager.Cmdlets.SdkModels;
+using ProjectResources = Microsoft.Azure.Commands.ResourceManager.Cmdlets.Properties.Resources;
+
+namespace Microsoft.Azure.Commands.ResourceManager.Cmdlets.Implementation
+{
+    /// <summary>
+    /// Deletes a deployment.
+    /// </summary>
+    [Cmdlet(VerbsCommon.Remove, "AzureRmDeployment", SupportsShouldProcess = true,
+        DefaultParameterSetName = RemoveAzureDeploymentCmdlet.DeploymentNameParameterSet), OutputType(typeof(bool))]
+    public class RemoveAzureDeploymentCmdlet : ResourceManagerCmdletBase
+    {
+        /// <summary>
+        /// The deployment Id parameter set.
+        /// </summary>
+        internal const string DeploymentIdParameterSet = "RemoveByDeploymentId";
+
+        /// <summary>
+        /// The deployment name parameter set.
+        /// </summary>
+        internal const string DeploymentNameParameterSet = "RemoveByDeploymentName";
+
+        /// <summary>
+        /// The input object parameter set.
+        /// </summary>
+        internal const string InputObjectParameterSet = "RemoveByInputObject";
+
+        [Alias("DeploymentName")]
+        [Parameter(Position = 0, ParameterSetName = RemoveAzureDeploymentCmdlet.DeploymentNameParameterSet, Mandatory = true,
+            ValueFromPipelineByPropertyName = true, HelpMessage = "The name of the deployment.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Alias("DeploymentId", "ResourceId")]
+        [Parameter(ParameterSetName = RemoveAzureDeploymentCmdlet.DeploymentIdParameterSet, Mandatory = true,
+            HelpMessage = "The fully qualified resource Id of the deployment. example: /subscriptions/{subId}/providers/Microsoft.Resources/deployments/{deploymentName}")]
+        [ValidateNotNullOrEmpty]
+        public string Id { get; set; }
+
+        [Parameter(ParameterSetName = RemoveAzureDeploymentCmdlet.InputObjectParameterSet, Mandatory = true, 
+            ValueFromPipeline = true, HelpMessage = "The deployment object.")]
+        public PSDeployment InputObject { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            ConfirmAction(
+                ProjectResources.DeleteDeploymentMessage,
+                Name,
+                () =>
+                {
+                    var deploymentName = !string.IsNullOrEmpty(this.Name)
+                        ? this.Name
+                        : !string.IsNullOrEmpty(this.Id) ? ResourceIdUtility.GetResourceName(this.Id) : this.InputObject.DeploymentName;
+
+                    ResourceManagerSdkClient.DeleteDeploymentAtSubscriptionScope(deploymentName);
+                    WriteObject(true);
```

Comment:
@Tiano2017 nit: to be consistent with other `Remove-*` cmdlets, you should add a [`-PassThru`](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/patterns-for-standard-cmdlets.md#c-example-2) parameter and write `true` when the user provides that parameter; otherwise, the cmdlet should return nothing

Reply:
I guess I'll leave it because Remove-AzureRmResourceGroupDeployment does the same.